myia/anf_ir.py

```diff
@@ -79,6 +82,26 @@ def __init__(self) -> None:
         self.return_: Apply = None
         self.debug = GraphDebug(self)
 
+    @property
+    def output(self) -> 'ANFNode':
+        """
+        Return the graph's output.
+
+        Equal to `self.return_.inputs[1]`, if it exists. Unlike `return_`,
+        `output' may be a constant or belong to a different graph.
+        """
+        if not self.return_ or len(self.return_.inputs) != 2:
+            raise Exception('Graph has no output.')
+        return self.return_.inputs[1]
+
+    @output.setter
+    def output(self, value: 'ANFNode') -> None:
+        """Set the graph's output."""
+        if self.return_:
+            self.return_.inputs[1] = value
+        else:
+            self.return_ = Apply([Constant(RETURN), value], self)
```

Comment:
There should probably be only one node `Constant(RETURN)` in the entire graph, not a different node for each function. One the other hand, we should be able to create multiple `Constant(RETURN)` objects if we want to keep two programs entirely separate (allowing us to e.g. look at `Constant(RETURN).uses` without mixing up two programs).

This is general thing: We don't want a new `Constant(1)` node each time the constant 1 is used somewhere. I am not entirely sure where this logic should be handled though...

Reply:
I don't think it's a major problem for now if we have multiple `Constant(RETURN)` for now.  We can push that for later.

Reply:
Yeah I was wondering about what to do about that, but I don't think merging constants and other nodes immediately is a particularly important or pressing issue.

Reply:
Okay, let's merge it then but please make an issue about duplicate nodes. Note that it's not just constants, any node whose `(inputs, value, graph)` values are identical shouldn't be duplicated (that's the idea of doing CSE at graph construction).